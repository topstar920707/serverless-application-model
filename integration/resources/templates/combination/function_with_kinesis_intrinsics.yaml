Parameters:
  IntValue:
    Type: Number
    Default: 100

  One:
    Type: Number
    Default: 1

  StartingPositionValue:
    Type: String
    Default: LATEST

  FunctionResponseTypesValue:
    Type: String
    Default: ReportBatchItemFailures

Conditions:
  TrueCondition:
    Fn::Equals:
      - true
      - true
  FalseCondition:
    Fn::Equals:
      - true
      - false

Resources:
  MyLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs12.x
      CodeUri: ${codeuri}
      MemorySize: 128

      Events:
        KinesisStream:
          Type: Kinesis
          Properties:
            BatchSize:
              Ref: IntValue
            BisectBatchOnFunctionError:
              Fn::If:
                - FalseCondition
                - True
                - False
            Enabled:
              Fn::If:
                - TrueCondition
                - True
                - False
            FunctionResponseTypes:
              - Ref: FunctionResponseTypesValue
            MaximumBatchingWindowInSeconds:
              Ref: One
            MaximumRecordAgeInSeconds:
              Ref: IntValue
            MaximumRetryAttempts:
              Ref: One
            ParallelizationFactor:
              Ref: One
            StartingPosition:
              Ref: StartingPositionValue
            Stream:
              # Connect with the stream we have created in this template
              Fn::Join:
                - ''
                - - 'arn:'
                  - Ref: AWS::Partition
                  - ':kinesis:'
                  - Ref: AWS::Region
                  - ':'
                  - Ref: AWS::AccountId
                  - ':stream/'
                  - Ref: MyStream
            TumblingWindowInSeconds:
              Ref: IntValue
  MyStream:
    Type: AWS::Kinesis::Stream
    Properties:
      ShardCount: 1
