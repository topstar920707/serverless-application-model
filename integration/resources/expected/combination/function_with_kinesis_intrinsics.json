[
  { "LogicalResourceId":"MyLambdaFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyLambdaFunctionRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"MyStream", "ResourceType":"AWS::Kinesis::Stream" },
  { "LogicalResourceId":"MyLambdaFunctionKinesisStream", "ResourceType":"AWS::Lambda::EventSourceMapping" }
]